src/ciel/plugin.go

```diff
@@ -0,0 +1,59 @@
+package main
+
+import (
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"syscall"
+)
+
+const (
+	PluginDir = LibExecDir + "/ciel-plugin/"
+)
+
+// Plugin defines a ciel plugin
+type Plugin struct {
+	Name string
+	// TODO: get script usage from header comments
+	Usage string
+}
+
+func cielPlugin() int {
+	proc := PluginDir + "ciel-" + SubCommand
+	cmd := exec.Command(proc, SubArgs...)
+	cmd.Stdin = os.Stdin
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	err := cmd.Run()
+	if err != nil {
+		if exitError, ok := err.(*exec.ExitError); ok {
+			exitStatus := exitError.Sys().(syscall.WaitStatus)
+			return exitStatus.ExitStatus()
+		}
+		log.Printf("failed to run plugin %s: %v\n", SubCommand, err)
+		return 1
+	}
+	return 0
+}
+
+func getPlugins() []Plugin {
+	var Plugins []Plugin
+	err := filepath.Walk(PluginDir, func(path string, f os.FileInfo, err error) error {
+		if f == nil {
+			return err
+		}
+		if f.IsDir() {
+			return nil
```

Comment:
That could be fixed by using a cached map of plugin name and abs path. And it would require a decent amount of resource to walk through that folder each time `ciel` is called. Or we can disable plugins in subdirectories. Depends on whether our developers needs that functionality.